### PR TITLE
[Partial] Adds questionnaires to Foomedical example  #4758

### PR DESCRIPTION
**Pull request by filipeximenes**
# Description
This PR adds:
- Page to respond the questionnaire (for a given `Questionnaire` id)
- A page in the Health Record listing all `QuestionnaireResponse`s
- A page to display answers to a given `QuestionnaireResponse`

The following items will be tackled in a separate PR:
- Define how users are going to access the page to respond the questionnaire
- Define what will be the default questionnaire 
- In the responses page filter the query by the default questionnaire so only responses to it are shown
- Improve displaying of questionnaire response items

# Steps to test

- [ ] Create a `Questionaire`
- [ ] Go to `/Questionnaire/:questionnaireId` fill questionnaire values and submit
- [ ] You should be redirected to `/health-record/questionnaire-responses/` and see a success toast
- [ ] Click one of the questionnaire response dates 
- [ ] You should see the answers to the questionnaire response

# Resources

https://github.com/medplum/medplum/assets/1180884/4e6f402d-4b7f-41c8-ae5a-8abd40bbf1e4

### CODE REVIEW

File: examples/foomedical/src/pages/health-record/Response.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { Box, Paper, Stack, Title, TitleOrder } from '@mantine/core';
+import { formatDate } from '@medplum/core';
+import { QuestionnaireResponseItem, QuestionnaireResponseItemAnswer } from '@medplum/fhirtypes';
+import { CodeableConceptDisplay, QuantityDisplay, RangeDisplay, useMedplum } from '@medplum/react';
+import { useParams } from 'react-router-dom';
+
+export function Response(): JSX.Element {
+  const medplum = useMedplum();
+  const { responseId } = useParams();
+  const questionnaireResponse = medplum.searchOne('QuestionnaireResponse', `_id=${responseId}`).read();
```
Comment (filipeximenes) [inline examples/foomedical/src/pages/health-record/Response.tsx:10]: Not using `useResource` here to keep it consistent with other health record pages and to make use of the `<Suspense />` loading in the root page. LMK if you'd prefer to keep it this way or update to the new standard.

File: examples/foomedical/src/pages/health-record/Responses.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { InfoButton } from '../../components/InfoButton';
+import { InfoSection } from '../../components/InfoSection';
+
+export function Responses(): JSX.Element {
+  const medplum = useMedplum();
+  const theme = useMantineTheme();
+  const navigate = useNavigate();
+  const profile = useMedplumProfile() as Patient;
+  const responses = medplum
+    .searchResources('QuestionnaireResponse', `source=${getReferenceString(profile)}&_sort=-authored`)
```
Comment (filipeximenes) [inline examples/foomedical/src/pages/health-record/Responses.tsx:16]: Same as in the other cases, making use of suspense for displaying the loading spinner

File: examples/foomedical/src/pages/health-record/Response.tsx
Comment (filipeximenes) [inline examples/foomedical/src/pages/health-record/Response.tsx:1]: As requested in the [issue](https://github.com/medplum/medplum/issues/4758) this page is based on this [existing implementation](https://github.com/medplum/medplum/blob/main/examples/medplum-chart-demo/src/components/EncounterNoteDisplay.tsx), I've done a bit of improvement by converting the functions to React components but this is still not in a good shape to be moved to the core library. Once we figure the design of how we want to display these I can propose a better API.

File: examples/foomedical/src/main.tsx [hunk 3]
```diff
@@ -40,6 +41,7 @@ root.render(
     <BrowserRouter>
       <MedplumProvider medplum={medplum}>
         <MantineProvider theme={theme}>
+          <Notifications />
```
Comment (filipeximenes) [inline examples/foomedical/src/main.tsx:44]: This was missing so notifications were not showing. For some reason the build is now breaking with the following error, I'll investigate what's going on.
```
Error: src/main.tsx(2,10): error TS2300: Duplicate identifier 'Notifications'.
Error: src/main.tsx(4,10): error TS2300: Duplicate identifier 'Notifications'.
```

File: examples/foomedical/src/pages/health-record/Response.tsx
Comment (rahul1) [reply examples/foomedical/src/pages/health-record/Response.tsx:1]: @filipeximenes , I agree with your comments. Right now we don't have a good design, but we will file this away for future work

**Review by rahul1 — APPROVED** [review]

**Pull request merged**